src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,176 @@
+                    method = "public/subscribe",
+                    params = SubscribeParams(channels = listOf("ticker.$instrumentName.$TICKER_SUBSCRIPTION_INTERVAL"))
+                )
+                send(Frame.Text(json.encodeToString(subscribeRequest)))
+                println("Sent subscription request: ${json.encodeToString(subscribeRequest)}")
+
+
+                incoming.consumeEach { frame ->
+                    if (frame is Frame.Text) {
+                        val text = frame.readText()
+                        try {
+                            val response = json.decodeFromString<DeribitResponse<TickerData>>(text)
+
+                            if (response.method == "subscription" && response.params?.channel == "ticker.$instrumentName.$TICKER_SUBSCRIPTION_INTERVAL") {
+                                val tickerData = response.params.data
+                                val pricePoint =
+                                    PricePoint(tickerData.timestamp, tickerData.markPrice)
+                                addPricePointToWindow(pricePoint)
+                            } else if (response.id != null && response.result != null) {
+                                println("Subscription confirmed for channels: ${response.result}")
+                            } else if (response.id != null && response.error != null) {
+                                System.err.println("Error from Deribit: ${response.error.message} (Code: ${response.error.code})")
+                            }
+                        } catch (e: Exception) {
+                            System.err.println("Error parsing WebSocket message: $text \n ${e.message}")
+                        }
+                    }
+                }
+            }
+        } catch (e: Exception) {
+            System.err.println("WebSocket connection error: ${e.message}")
+            e.printStackTrace()
+        } finally {
+            client.close()
+            println("WebSocket client closed.")
+        }
+    }
+
+    joinAll(wsJob, trendJob)
+}
+
+fun addPricePointToWindow(pricePoint: PricePoint) {
+    synchronized(priceDataWindow) {
+        priceDataWindow.addLast(pricePoint)
+        // Remove old data points
+        val windowLimitTimestamp = pricePoint.timestamp - (SLIDING_WINDOW_SECONDS * 1000)
+        while (priceDataWindow.isNotEmpty() && priceDataWindow.first().timestamp < windowLimitTimestamp) {
+            priceDataWindow.removeFirst()
+        }
+    }
+}
+
+
+fun calculateLinearTrend(points: List<PricePoint>): Pair<Double, Double> {
+    if (points.size < 2) return Pair(0.0, points.firstOrNull()?.price ?: 0.0)
+
+    val n = points.size.toDouble()
+    val firstTimestamp = points.first().timestamp
+
+    val x = points.map { (it.timestamp - firstTimestamp).toDouble() / 1000.0 } // Relative time in seconds
+    val y = points.map { it.price }
+
+    val sumX = x.sum()
+    val sumY = y.sum()
+    val sumXY = x.zip(y).sumOf { it.first * it.second }
```

Comment:
Просто предложение, чтобы сделать чуть меньше боксинга:
```suggestion
    val sumXY = x.zip(y) { x, y -> x * y }.sum()
```